src/map.rs

```diff
@@ -192,8 +185,20 @@ impl Map {
         parse_tag!(parser, "map", {
             "tileset" => |attrs| {
                 let res = Tileset::parse_xml_in_map(parser, attrs, map_path)?;
-                tileset_cache.get_or_insert_tileset(&res.tileset_path, res.tileset);
-                tilesets.push(TilesetRef{first_gid: res.first_gid, path: res.tileset_path});
+                let path = match res.result_type {
+                    EmbeddedParseResultType::ExternalReference { tileset_path } => {
+                        let path = crate::ResourcePath::External{ path: tileset_path.clone() };
+                        let file = File::open(&tileset_path).map_err(|err| TiledError::CouldNotOpenFile{path: tileset_path.clone(), err })?;
+                        tileset_cache.get_or_try_insert_tileset_with(path.clone(), || Tileset::new_external(file, &tileset_path))?;
+                        path
+                    }
+                    EmbeddedParseResultType::Embedded { tileset } => {
+                        let path = ResourcePath::Embedded { container_path: map_path.to_owned(), index: tilesets.len()};
+                        tileset_cache.get_or_insert_tileset(path.clone(), tileset);
```

Comment:
I don't really understand this. Why did you introduce `ResourcePath::Embedded`? What's the point of caching embedded tilesets, given that no other map could reference them?